lib/util/check-token.ts

```diff
@@ -0,0 +1,51 @@
+import { PlatformId } from '../constants';
+import { logger } from '../logger';
+import { GithubReleasesDatasource } from '../modules/datasource/github-releases';
+import { GithubTagsDatasource } from '../modules/datasource/github-tags';
+import type { PackageFile } from '../modules/manager/types';
+import * as memCache from '../util/cache/memory';
+import * as hostRules from './host-rules';
+
+export function checkGithubToken(
+  packageFiles: Record<string, PackageFile[]> | undefined
+): void {
+  const { token } = hostRules.find({
+    hostType: PlatformId.Github,
+    url: 'https://api.github.com',
+  });
+
+  if (token) {
+    logger.trace('GitHub token is found');
+    return;
+  }
+
+  const githubDeps: string[] = [];
+  for (const files of Object.values(packageFiles ?? {})) {
+    for (const file of files ?? []) {
+      for (const dep of file.deps ?? []) {
+        if (
+          dep.datasource === GithubTagsDatasource.id ||
+          dep.datasource === GithubReleasesDatasource.id
+        ) {
+          dep.skipReason = 'github-token-required';
+          if (dep.depName) {
+            githubDeps.push(dep.depName);
+          }
+        }
+      }
+    }
+  }
+
+  if (githubDeps.length > 0) {
+    const warningLogged = memCache.get<boolean | undefined>(
+      'github-token-required-warning-logged'
+    );
+    if (!warningLogged) {
+      logger.warn(
```

Comment:
Would it be possible to wrap this in https://github.com/renovatebot/renovate/pull/14902/files#diff-420a4cb0e5c3eccd3c0e4fd5748f499dcfdae5f7ef51b40249ac10ff5331e9d0R377 like we did for the others? When this change went in, we're now seeing a warning again in the Dependency Dashboard, which we'd like to suppress. Thanks!

Reply:
cc @zharinov @viceice @rarkins

Reply:
please open a new discussion